recipes/pcl/all/conanfile.py

```diff
@@ -104,102 +105,83 @@ def requirements(self):
+        copy(self, "LICENSE.txt",
+             dst=os.path.join(self.package_folder, "licenses"),
+             src=self.source_folder)
+
         cmake = CMake(self)
         cmake.install()
 
         rmdir(self, os.path.join(self.package_folder, "cmake"))
         rmdir(self, os.path.join(self.package_folder, "share"))
-        # For the legal reasons, and in order to reduce the size of packages, it's not allowed to package Microsoft Visual Studio runtime libraries, such as msvcr80.dll, msvcp80.dll, vcruntime140.dll and so on.
-        # See https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#kb-h021-ms-runtime-files
-        rmdir(self, os.path.join(self.package_folder, "bin"))
-
-    @property
-    def _pcl_lib_components(self):
-        def usb():
-            return ["libusb::libusb"] if self.options.with_libusb else []
-        def png():
-            return ["libpng::libpng"] if self.options.with_png else []
-        def qhull():
-            return ["qhull::qhull"] if self.options.with_qhull else []
-
-        return {
-            "common": {"requires": ["eigen::eigen3", "boost::boost"]},
-            "kdtree": {"requires": ["common", "flann::flann"]},
-            "octree": {"requires": ["common"]},
-            "search": {"requires": ["common", "kdtree", "octree", "flann::flann"]},
-            "sample_consensus": {"requires": ["common", "search"]},
-            "filters": {"requires": ["common", "sample_consensus", "search", "kdtree", "octree"]},
-            "2d": {"requires": ["common", "filters"], "header_only": True},
-            "geometry": {"requires": ["common"], "header_only": True},
-            "io": {"requires": ["common", "octree", "zlib::zlib"] + png() + usb(), "extra_libs": ["io_ply"]},
-            "features": {"requires": ["common", "search", "kdtree", "octree", "filters", "2d"]},
-            "ml": {"requires": ["common"]},
-            "segmentation": {"requires": ["common", "geometry", "search", "sample_consensus", "kdtree", "octree", "features", "filters", "ml"]},
-            "surface": {"requires": ["common", "search", "kdtree", "octree"] + qhull()},
-            "registration": {"requires": ["common", "octree", "kdtree", "search", "sample_consensus", "features", "filters"]},
-            "keypoints": {"requires": ["common", "search", "kdtree", "octree", "features", "filters"]},
-            "tracking": {"requires": ["common", "search", "kdtree", "filters", "octree"]},
-            "recognition": {"requires": ["common", "io", "search", "kdtree", "octree", "features", "filters", "registration", "sample_consensus", "ml"]},
-            "stereo": {"requires": ["common", "io"]}
-        }
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        # Remove MSVC runtime libraries
+        for dll_pattern_to_remove in ["concrt*.dll", "msvcp*.dll", "vcruntime*.dll"]:
+            rm(self, dll_pattern_to_remove, os.path.join(self.package_folder, "bin"))
```

Comment:
Indeed my cleanup was not enough restrictive, and is the cause of the last failing pipeline.
I'm not a Windows expert and doesn't understand why windows dll are installed in bin folder.
Another thing I don't understand is why previous conan v2 pipeline (https://github.com/conan-io/conan-center-index/pull/18389#issuecomment-1670332274) failed with following error message while I have only changed pcl files "Changes not allowed in build 7 : [.c3i/conan_v2_ready_references.yml, recipes/pcl/all/conandata.yml, recipes/pcl/all/conanfile.py, recipes/pcl/all/patches/0001-Add-Eigen3-Eigen-target-in-pcl_common-target.patch, recipes/pcl/all/patches/0001-KB-H020_avoid_pkgconfig_files.patch, recipes/pcl/all/patches/0001-ReportFixAboutMemoryConsumptionDuringBuild.patch, recipes/pcl/all/patches/0001-cmake_use_requirements_name_from_cmakedeps.patch, recipes/pcl/all/test_package/CMakeLists.txt, recipes/pcl/all/test_package/conanfile.py, recipes/pcl/all/test_package/test_package.cpp, recipes/pcl/config.yml]"